File: CipherSample.java
import java.util.Base64;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

public class CipherSample {

  private static final String KEY_SPEC = "AES";

  private static final String CIPHER_MODE = "AES/ECB/PKCS5Padding";

  public static void main(String[] args) {
    String key = "abcdefghijklmnop"; // <-128bit = 16文字のキー
    byte[] keyBytes = key.getBytes(); // バイト列に変換

    CipherSample cs = new CipherSample();

    byte[] enBytes = cs.encryptECB("hello".getBytes(), keyBytes);

     // Base64して表示
    System.out.println(Base64.getEncoder().encodeToString(enBytes));

    byte[] deBytes = cs.decryptECB(enBytes, keyBytes);

    System.out.println(new String(deBytes));

  }


  public byte[] encryptECB(byte[] data, byte[] secret_key) {

    SecretKeySpec sKey = new SecretKeySpec(secret_key, KEY_SPEC);
    try {
      Cipher cipher = Cipher.getInstance(CIPHER_MODE);
      cipher.init(Cipher.ENCRYPT_MODE, sKey);
      return cipher.doFinal(data);

    } catch (Exception e) {
      e.printStackTrace();
    }
    return new byte[0];
  }

  public byte[] decryptECB(byte[] encData, byte[] secret_key) {

    SecretKeySpec sKey = new SecretKeySpec(secret_key, KEY_SPEC);
    try {
      Cipher cipher = Cipher.getInstance(CIPHER_MODE);
      cipher.init(Cipher.DECRYPT_MODE, sKey);
      return cipher.doFinal(encData);
    } catch (Exception e) {
      e.printStackTrace();
    }
    return new byte[0];
  }
}

File: CipherSample.rb
require 'openssl'
require 'base64'

# 暗号化
def encrypt(data, key, iv)
	enc = OpenSSL::Cipher.new("AES-128-ECB")
	enc.encrypt

	enc.key = key
	enc.iv = iv   # 初期化ベクトルは不要だが、エラー回避のためにセット

	# 暗号化
	encrypted_data = ""
	encrypted_data << enc.update(data)
	encrypted_data << enc.final

	return encrypted_data
end

# 復号化
def decrypt(data, key, iv)
	dec = OpenSSL::Cipher.new("AES-128-ECB")
	dec.decrypt

	dec.key = key
	dec.iv = iv   # 初期化ベクトルは不要だが、エラー回避のためにセット

	# 復号化
	decrypted_data = ""
	decrypted_data << dec.update(data)
	decrypted_data << dec.final

	return decrypted_data
end


data = "hello"
key = "abcdefghijklmnop"
iv  = "0000000000000000"

p "Original: " + data
encrypted_data = encrypt(data, key, iv)
p "Encript : " + Base64.encode64(encrypted_data)
p "Decript : " + decrypt(encrypted_data, key, iv)